src/black/strings.py

```diff
@@ -236,3 +247,31 @@ def normalize_string_quotes(s: str) -> str:
         return s  # Prefer double quotes
 
     return f"{prefix}{new_quote}{new_body}{new_quote}"
+
+
+def normalize_unicode_escape_sequences(leaf: Leaf) -> None:
+    """Replace hex codes in Unicode escape sequences with lowercase representation."""
+    text = leaf.value
+    prefix = get_string_prefix(text)
+
+    def replace(m: Match[str]) -> str:
+        groups = m.groups()
+        back_slashes = groups[0]
+
+        if len(back_slashes) % 2 == 0 or prefix == "r":
```

Comment:
This raw string check is too strict, you need to check for a `r` in a lowercased prefix, otherwise both of these cases crash with an AST safety error:

```python
no_touch = R"\xA1"
```

```python
no_touch = fr"\xA1"
```

Reply:
I'll note that this would be unnecessary if the other r-prefix related suggestion was implemented.

Reply:
Indeed, removed this branch